### DISCUSSION TITLE
ChatGPT knows about Stacked Git

### DISCUSSION THREAD

**Discussion by NonLogicalDev** [stacked-git/stgit#244] [General]
![image](https://user-images.githubusercontent.com/1669886/206879468-3fafc5ca-d004-42fa-ba6c-7860478feeda.jpeg)

**Comment by ctmarinas** [comment]
Well, ChatGPT needs to read a bit more on the history ;). Stacked Git was largely based on the Quilt patch management tool (which I was actively using at the time). The first RFC turned up a couple of months after Git was announced. IIRC the Mercurial Queues feature was added much later.

**Reply by NonLogicalDev** [reply]
@ctmarinas the legend themselves =].
Thank you so much for founding this project!

This is totally fascinating, I always thought that Quilt was an original inspiration for MQ, and StGit was inspired by both the MQ and Quilt. But seems like I indeed got it backwards.

Seems like the timeline is:

* 2003 (Jan 21) - Quilt First commit

  * https://github.com/yashi/quilt/commit/4d11e99bfc25e0261111d202a3a2afdf97daea5c

* 2005 (Apr 07)- Initial version of Git Released

  * https://github.com/git/git/commit/e83c5163316f89bfbde7d9ab23ca2e25604af290

* 2005 (Jul 10) - Initial StGit version published (0.4)

  * https://github.com/stacked-git/stgit/commit/41a6d8591d5962dbfe8e372fff10c60e06718083

* 2006 (Feb 28) - Mercurial Queues extension gets added to Mercurial repo

  * https://github.com/dscho/hg/commit/9ce23d53941fd5b2aced854ec9fcac13e4f10fea

@ctmarinas what was the history behind StGits creation? Did MQ in fact got its inspiration from not only Quilt but Stacked Git as well, was there any mutual inspiration as projects grew?
 
Also curious which RFC did you reference?

**Reply by ctmarinas** [reply]
By RFC I meant StGit 0.1 (I had forgotten whether I put an RFC in the subject or not). Digging the mailing list archives, I found the first post - https://lore.kernel.org/all/tnxy899zzu7.fsf@arm.com/. I wanted something that allowed me to modify patches following review and Git didn't have an interactive 'rebase' command. Quilt was the closest to what I needed, though at the time I was a bit fascinated by 'darcs' and its patch theory (you can see a bit of that in the cover letter above, trying to "formalise" what a patch is ;)). Unfortunately darcs did not scale to the Linux kernel source. Not sure whether MQ got its inspiration from Quilt or StGit ('git rebase -i' can also be used to modify patches, I don't remember when the interactive option was added). But what's clear is that such feature was needed by many projects used to generating patches for review, modifying them and so on.
